weave/trace/weave_client.py

```diff
@@ -2111,4 +2196,71 @@ def elide_display_name(name: str) -> str:
     return name
 
 
-__docspec__ = [WeaveClient, Call, CallsIter]
+class ObjectVersionCollection:
+    """A collection of object versions for a single object.
```

Comment:
The lazy loading seems to be a key motivation for this class, I'd mention it in here.

Reply:
done